### ISSUE TITLE
unidirectional OneToMany

### ISSUE THREAD

**Issue by fixako** [typeorm/typeorm#3233]
**Issue type:**

[ ] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

Do you plan adding unidirectional OneToMany support? If yes when?

**Comment by pleerock** [comment]
having unidirectional OneToMany means we have to define **inverse table's column** inside some other entities. This can create a maintainece hell for that table and its entity. So, it really looks like a bad pattern + its going to introduce complexity. So, no, we don't have such plans.

The only reason to prefer to have OneToMany without ManyToOne is to keep entities cleaner. If you want that, I suggest you to [separate your entity schema definitions](http://typeorm.io/#/separating-entity-definition) from your models, this will make your models cleaner and separate concerns.

**Comment by ekifox** [comment]
Why? For example we have SportsGame and News tables.
News can be added without tagging with any game, but after, when some game is added in database, administrators can tag a few news posts for a that game. But news doesn't need to know in that game it has been used.
One match -> Few news, without inverse, only one way.

**Comment by ekifox** [comment]
Example 2: Streamers and Matches

We have the streamer "SportsOne" (id: 1) and the games "Barcelona - Juventus" and "Real Madrid - Valencia"

The "SportsOne" today can stream "Barcelona - Juventus":
`Match{ team1: 'Barcelona', team2: 'Juventus', streamersId[]: [1] }`

Tomorrow "SportsOne" wants to stream "Real Madrid - Valencia":
`Match{ team1: 'Real Madrid', team2: 'Valencia', streamersId[]: [1] }`

We handle information about streamers of each match for a years, but streamers does not change any of their states in their database table.

In short, streamers in my database do not need to keep back links to the matches that they streamed.

**Comment by ekifox** [comment]
@pleerock please add unidirectional support for OneToMany

**Comment by kaskrish** [comment]
Without unidirectional support, the domain model will have to bear orm impedance mismatch, as unnecessary associations have to be maintained..

**Comment by pleerock** [comment]
I understand your purposes since I had same purposes a long time ago before even creating typeorm. However your absolute goal of domain separation is to make application more maintainable, but this won't do it, and instead, will harm you. Just imagine - you have entity `User` and anyone can add columns into its table - and try to guess who is doing that! Module separation doesn't make sense when it brings you implicit behaviour and increases maintenance. 

If you are building a really large scale application I suggest you to do it following way:

* `entities` - special module for entity schemas
* `user` - user module with its own models
* `photo` - photo module with its own models
* `...` - other modules with their own models

This way you'll really make your app scalable and each module will stay in its own scope. You can even make microservice architecture this way and deploy any module together with entities module when you apply changes to it. Treat entities as a separate module with its own responsibility to define a database structure, don't treat it as a place for your domain models. Entity schemas work really great in such scenarios.

We are using a relational database, and relational database is all about coupling between tables, and this enforces you to do same in your code unfortunately.

**Comment by nhh** [comment]
@pleerock You are missing parent child relations here. It's completely fine for a user, only existing as a child entity, and being persisted from the parent entity. You are lifting all entities to the root level, which is simply not how it works. Any assumption like "will harm you", "Just imagine - you have entity User and anyone can add columns into its table"  should be avoided for established patterns. Using unidirectional relations is not bad per se, it's a tradeoff.

So, please add unidirectional @OneToMany support. Thank you

**Comment by mwaeckerlin** [comment]
This is a very important feature, because it would solve the problem described in #5142

**Comment by mwaeckerlin** [comment]
There is another very good reason for this feature;

- Three Tables: A, B, C
- One B owns many As
- One C owns many As

So A is sometimes owned by B, sometimes by C; there is no way to link back!

**Comment by mwaeckerlin** [comment]
Another reason for this feature:

I have a price table. Each price can be associated to zero or many groups of users that pay this price.

The price table has a `@OneToMany` relation to the group tablle. But the group has absolutely nothing to to with the price and there must not by any price key in the group table. A group has nothing to with a price, and in fact there are many groups with no price relation at all. Just a price can be limited to some groups. So from the logic, this correlation is completely unidirectional.

**Comment by johnlister** [comment]
> There is another very good reason for this feature;
> 
> * Three Tables: A, B, C
> * One B owns many As
> * One C owns many As
> 
> So A is sometimes owned by B, sometimes by C; there is no way to link back!

We have exactly the same issue and for me this is a common pattern. I don't understand this comment: "**having unidirectional OneToMany means we have to define inverse table's column**", all you need to do is define the joinColumn on the oneToMany relation and TypeORM can work out the join condition. If we are missing the manyToOne relation, you don't need to set up any foreign key relationship in the database or create a column in the inverse table - I would assume the developer has already done that. Nearly every other ORM does it this way.

**Comment by skobaken7** [comment]
So dirty workaround :joy: 
``` typescript
@Entity()
class User {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @OneToMany(() => Photo, (photo) => photo.dummyFieldForManyToOne, { eager: true })
  photos: Photo[];
}

@Entity()
class Photo {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  userId: number;

  @Column()
  title: string;

  @ManyToOne(() => Photo)
  @JoinColumn({ name: 'user_id' })
  dummyFieldForManyToOne: unknown;
}
```

**Comment by mujehoxe** [comment]
I think this is a very useful feature since it might reduce coupling between the two entities, therefore make the second more reusable.

**Comment by hanspoo** [comment]
Unidirectional association patterns are completely valid in all types of models: UML, relational, human, etc, they solve a real situation and are widely used, thanks for your time taken with this software i appreciate that, but you are wrong trying to impose your view to the rest, please read the posts..you are alone in this,  if it's too hard for you implement this, be honest, shout at us to make it, its your right when you give your time, that will be better than trying to explain the inexplicable.

**Comment by IgorArnaut** [comment]
Yeah. I have these models: https://pastebin.com/6dmxBcnN.

BiDirectional one-to-many relations would make things complicated, especially when there's model inheritance.

**Comment by IgorArnaut** [comment]
> Yeah. I have these models: https://pastebin.com/5d5mvPgF.
> 
> BiDirectional one-to-many relations would make things complicated, especially when there's model inheritance.

I'd have to make `ApartmentAddress`, `ApartmentBuilding`, `ApartmentItem`, `ApartmentTerms`, `ApartmentListing` for `Apartment` and `LocalAddress`, `LocalBuilding`, `LocalItem`, `LocalTerms`, `LocalListing` for `Local` Both `Apartment` and `Local` extend `Realty`. With bi-directional one-to-many I'd have to make these unecessary additional models just for those `Realty` children. It rather have a single `Address`, single `Building`, single `Terms`, single `Listing` and single `Item` models. Do you see the problem here???

**Comment by IArnaut2** [comment]
> > Yeah. I have these models: https://pastebin.com/5d5mvPgF.
> > BiDirectional one-to-many relations would make things complicated, especially when there's model inheritance.
> 
> I'd have to make `ApartmentAddress`, `ApartmentBuilding`, `ApartmentItem`, `ApartmentTerms`, `ApartmentListing` for `Apartment` and `LocalAddress`, `LocalBuilding`, `LocalItem`, `LocalTerms`, `LocalListing` for `Local` Both `Apartment` and `Local` extend `Realty`. With bi-directional one-to-many I'd have to make these unecessary additional models just for those `Realty` children. It rather have a single `Address`, single `Building`, single `Terms`, single `Listing` and single `Item` models. Do you see the problem here???

This is what I mean.
https://github.com/typeorm/typeorm/issues/5124#issuecomment-2835847707

How come nobody even bothered to see these arguments?
